In [ ]:
import numpy as np
import matplotlib.pyplot as plt
import my_submission as ms

Using TensorFlow backend.


input_trainset:  (33149, 28, 28, 1)
target_trainset:  (33149,)
input_testset1:  (8288, 28, 28, 1)
target_testset1:  (8288,)
input_testset2:  (28563, 28, 28, 1)
target_testset2:  (28563,)
input_testset3:  (36851, 28, 28, 1)
target_testset3:  (36851,)
--------------------
training_pairs:  (60156, 2, 28, 28, 1)
training_target:  (60156,)
--------------------
test_pairs_set1:  (15576, 2, 28, 28, 1)
test_target_set1:  (15576,)
--------------------
test_pairs_set2:  (54592, 2, 28, 28, 1)
test_target_set2:  (54592,)
--------------------
test_pairs_set3:  (25960, 2, 28, 28, 1)
test_target_set3:  (25960,)
--------------------
Train on 60156 samples, validate on 15576 samples
Epoch 1/5
12160/60156 [=====>........................] - ETA: 4:20 - loss: 0.7767 - accuracy: 0.7314

In [ ]:
#from tensorflow.python.client import device_lib
#print(device_lib.list_local_devices())

In [ ]:
#ms.save_mnist_dataset()
# Loads the already preprocessed dataset.
#(x_train, y_train), (x_test, y_test) = ms.load_mnist_dataset()
#print('x_train : ', x_train.shape, x_train.dtype)
#print('y_train : ', y_train.shape, y_train.dtype)
#print('x_test : ', x_test.shape, x_test.dtype)
#print('y_test : ', y_test.shape, y_test.dtype)

#x_train, y_train, x_test, y_test = ms.preprocess_mnist_dataset(x_train, y_train, x_test, y_test)
# Saves the preprocessed dataset locally as a .npz file.
#np.savez('mnist_dataset.npz', x_train = x_train, y_train = y_train, x_test = x_test, y_test = y_test)


In [ ]:
# Verifying that the split has been done correctly
#print("Unique numbers in train set (shouldn't be any of the 0,1,8 or 9):")
#unique, counts = np.unique(y_train, return_counts=True)
#dict(zip(unique, counts))

In [ ]:
# Verifying that the split has been done correctly
#print("Unique numbers in test set:")
#unique, counts = np.unique(y_test, return_counts=True)
#dict(zip(unique, counts))

In [ ]:
#ms.build_CNN(x_train, y_train, x_test, y_test)
#ms.siamese_network()

<h1>Results:</h1>
<p>
input_trainset:  (33149, 28, 28, 1)
target_trainset:  (33149,)
input_testset1:  (8288, 28, 28, 1)
target_testset1:  (8288,)
input_testset2:  (28563, 28, 28, 1)
target_testset2:  (28563,)
input_testset3:  (36851, 28, 28, 1)
target_testset3:  (36851,)
--------------------
training_pairs:  (60792, 2, 28, 28, 1)
training_target:  (60792,)
--------------------
test_pairs_set1:  (14940, 2, 28, 28, 1)
test_target_set1:  (14940,)
--------------------
test_pairs_set2:  (54592, 2, 28, 28, 1)
test_target_set2:  (54592,)
--------------------
test_pairs_set3:  (24900, 2, 28, 28, 1)
test_target_set3:  (24900,)
--------------------
Train on 60792 samples, validate on 14940 samples
Epoch 1/30
60792/60792 [==============================] - 77s 1ms/step - loss: 0.4699 - accuracy: 0.8709 - val_loss: 0.3246 - val_accuracy: 0.9457
Epoch 2/30
60792/60792 [==============================] - 76s 1ms/step - loss: 0.3249 - accuracy: 0.9497 - val_loss: 0.3058 - val_accuracy: 0.9588
Epoch 3/30
60792/60792 [==============================] - 77s 1ms/step - loss: 0.3172 - accuracy: 0.9535 - val_loss: 0.3049 - val_accuracy: 0.9585
Epoch 4/30
60792/60792 [==============================] - 77s 1ms/step - loss: 0.3120 - accuracy: 0.9568 - val_loss: 0.3056 - val_accuracy: 0.9592
Epoch 5/30
60792/60792 [==============================] - 77s 1ms/step - loss: 0.3090 - accuracy: 0.9582 - val_loss: 0.3009 - val_accuracy: 0.9617
Epoch 6/30
60792/60792 [==============================] - 77s 1ms/step - loss: 0.3080 - accuracy: 0.9586 - val_loss: 0.3017 - val_accuracy: 0.9616
Epoch 7/30
60792/60792 [==============================] - 77s 1ms/step - loss: 0.3060 - accuracy: 0.9601 - val_loss: 0.3001 - val_accuracy: 0.9622
Epoch 8/30
60792/60792 [==============================] - 77s 1ms/step - loss: 0.3051 - accuracy: 0.9604 - val_loss: 0.3027 - val_accuracy: 0.9615
Epoch 9/30
60792/60792 [==============================] - 77s 1ms/step - loss: 0.3050 - accuracy: 0.9604 - val_loss: 0.3013 - val_accuracy: 0.9616
Epoch 10/30
60792/60792 [==============================] - 77s 1ms/step - loss: 0.3033 - accuracy: 0.9613 - val_loss: 0.2996 - val_accuracy: 0.9629
Epoch 11/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.3031 - accuracy: 0.9613 - val_loss: 0.3032 - val_accuracy: 0.9603
Epoch 12/30
60792/60792 [==============================] - 77s 1ms/step - loss: 0.3027 - accuracy: 0.9616 - val_loss: 0.3006 - val_accuracy: 0.9624
Epoch 13/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.3017 - accuracy: 0.9622 - val_loss: 0.2994 - val_accuracy: 0.9626
Epoch 14/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.3011 - accuracy: 0.9625 - val_loss: 0.3011 - val_accuracy: 0.9616
Epoch 15/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.3008 - accuracy: 0.9626 - val_loss: 0.2983 - val_accuracy: 0.9635
Epoch 16/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.3002 - accuracy: 0.9630 - val_loss: 0.2977 - val_accuracy: 0.9639
Epoch 17/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.3010 - accuracy: 0.9628 - val_loss: 0.2985 - val_accuracy: 0.9638
Epoch 18/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.2998 - accuracy: 0.9633 - val_loss: 0.2977 - val_accuracy: 0.9640
Epoch 19/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.2994 - accuracy: 0.9633 - val_loss: 0.2960 - val_accuracy: 0.9645
Epoch 20/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.2993 - accuracy: 0.9636 - val_loss: 0.2974 - val_accuracy: 0.9638
Epoch 21/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.3003 - accuracy: 0.9631 - val_loss: 0.2977 - val_accuracy: 0.9639
Epoch 22/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.2997 - accuracy: 0.9635 - val_loss: 0.2969 - val_accuracy: 0.9640
Epoch 23/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.3007 - accuracy: 0.9630 - val_loss: 0.2992 - val_accuracy: 0.9628
Epoch 24/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.2998 - accuracy: 0.9632 - val_loss: 0.3008 - val_accuracy: 0.9626
Epoch 25/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.2994 - accuracy: 0.9636 - val_loss: 0.2985 - val_accuracy: 0.9635
Epoch 26/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.2993 - accuracy: 0.9636 - val_loss: 0.2978 - val_accuracy: 0.9633
Epoch 27/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.2988 - accuracy: 0.9638 - val_loss: 0.2976 - val_accuracy: 0.9637
Epoch 28/30
60792/60792 [==============================] - 78s 1ms/step - loss: 0.2983 - accuracy: 0.9638 - val_loss: 0.2995 - val_accuracy: 0.9633
Epoch 29/30
60792/60792 [==============================] - 1021s 17ms/step - loss: 0.2987 - accuracy: 0.9639 - val_loss: 0.2992 - val_accuracy: 0.9635
Epoch 30/30
60792/60792 [==============================] - 77s 1ms/step - loss: 0.2990 - accuracy: 0.9637 - val_loss: 0.2977 - val_accuracy: 0.9639
-------------------------------------------------------------------------------
----------------- THIS DATA WILL PROBABLY NOT BE USED -------------------------
Final accuracies for the different datasets using validation set number 1 after a total of 30 epoch(s).
* Accuracy on training set: 96.58%
* Accuracy on test set 1: 96.39%
* Accuracy on test set 2: 64.13%
* Accuracy on test set 3: 81.03%
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
Train on 60792 samples, validate on 54592 samples
Epoch 1/30
60792/60792 [==============================] - 92s 2ms/step - loss: 0.6636 - accuracy: 0.8179 - val_loss: 0.8833 - val_accuracy: 0.7039
Epoch 2/30
60792/60792 [==============================] - 91s 1ms/step - loss: 0.5633 - accuracy: 0.8824 - val_loss: 0.8144 - val_accuracy: 0.7333
Epoch 3/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5548 - accuracy: 0.8882 - val_loss: 0.8340 - val_accuracy: 0.7309
Epoch 4/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5491 - accuracy: 0.8919 - val_loss: 0.8045 - val_accuracy: 0.7442
Epoch 5/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5462 - accuracy: 0.8926 - val_loss: 0.9185 - val_accuracy: 0.6976
Epoch 6/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5445 - accuracy: 0.8941 - val_loss: 0.8937 - val_accuracy: 0.7216
Epoch 7/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5419 - accuracy: 0.8951 - val_loss: 0.8148 - val_accuracy: 0.7406
Epoch 8/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5424 - accuracy: 0.8951 - val_loss: 0.8815 - val_accuracy: 0.7179
Epoch 9/30
60792/60792 [==============================] - 91s 1ms/step - loss: 0.5413 - accuracy: 0.8960 - val_loss: 0.9073 - val_accuracy: 0.7057
Epoch 10/30
60792/60792 [==============================] - 91s 1ms/step - loss: 0.5402 - accuracy: 0.8964 - val_loss: 0.8569 - val_accuracy: 0.7081
Epoch 11/30
60792/60792 [==============================] - 91s 1ms/step - loss: 0.5404 - accuracy: 0.8964 - val_loss: 0.9386 - val_accuracy: 0.7026
Epoch 12/30
60792/60792 [==============================] - 91s 1ms/step - loss: 0.5398 - accuracy: 0.8969 - val_loss: 0.9338 - val_accuracy: 0.7011
Epoch 13/30
60792/60792 [==============================] - 91s 1ms/step - loss: 0.5386 - accuracy: 0.8973 - val_loss: 0.8804 - val_accuracy: 0.7180
Epoch 14/30
60792/60792 [==============================] - 4552s 75ms/step - loss: 0.5398 - accuracy: 0.8967 - val_loss: 0.8787 - val_accuracy: 0.7134
Epoch 15/30
60792/60792 [==============================] - 89s 1ms/step - loss: 0.5389 - accuracy: 0.8972 - val_loss: 0.9015 - val_accuracy: 0.7033
Epoch 16/30
60792/60792 [==============================] - 88s 1ms/step - loss: 0.5386 - accuracy: 0.8973 - val_loss: 0.9438 - val_accuracy: 0.7068
Epoch 17/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5378 - accuracy: 0.8977 - val_loss: 0.8121 - val_accuracy: 0.7401
Epoch 18/30
60792/60792 [==============================] - 89s 1ms/step - loss: 0.5391 - accuracy: 0.8973 - val_loss: 0.7806 - val_accuracy: 0.7636
Epoch 19/30
60792/60792 [==============================] - 89s 1ms/step - loss: 0.5383 - accuracy: 0.8976 - val_loss: 0.9030 - val_accuracy: 0.7194
Epoch 20/30
60792/60792 [==============================] - 89s 1ms/step - loss: 0.5374 - accuracy: 0.8978 - val_loss: 0.8644 - val_accuracy: 0.7217
Epoch 21/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5380 - accuracy: 0.8977 - val_loss: 0.8347 - val_accuracy: 0.7243
Epoch 22/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5379 - accuracy: 0.8977 - val_loss: 0.8830 - val_accuracy: 0.7214
Epoch 23/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5381 - accuracy: 0.8977 - val_loss: 1.0105 - val_accuracy: 0.6939
Epoch 24/30
60792/60792 [==============================] - 89s 1ms/step - loss: 0.5376 - accuracy: 0.8977 - val_loss: 0.9599 - val_accuracy: 0.7033
Epoch 25/30
60792/60792 [==============================] - 89s 1ms/step - loss: 0.5381 - accuracy: 0.8977 - val_loss: 0.8518 - val_accuracy: 0.7398
Epoch 26/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5382 - accuracy: 0.8975 - val_loss: 0.9031 - val_accuracy: 0.7273
Epoch 27/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5381 - accuracy: 0.8977 - val_loss: 0.9445 - val_accuracy: 0.7031
Epoch 28/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5377 - accuracy: 0.8977 - val_loss: 0.9105 - val_accuracy: 0.6965
Epoch 29/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5369 - accuracy: 0.8982 - val_loss: 0.9030 - val_accuracy: 0.7200
Epoch 30/30
60792/60792 [==============================] - 90s 1ms/step - loss: 0.5381 - accuracy: 0.8978 - val_loss: 0.9719 - val_accuracy: 0.7067
-------------------------------------------------------------------------------
----------------- THIS DATA WILL PROBABLY NOT BE USED -------------------------
Final accuracies for the different datasets using validation set number 2 after a total of 30 epoch(s).
* Accuracy on training set: 90.00%
* Accuracy on test set 1: 89.19%
* Accuracy on test set 2: 70.67%
* Accuracy on test set 3: 79.96%
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
Train on 60792 samples, validate on 24900 samples
Epoch 1/30
60792/60792 [==============================] - 86s 1ms/step - loss: 0.5435 - accuracy: 0.8574 - val_loss: 0.7158 - val_accuracy: 0.7788
Epoch 2/30
60792/60792 [==============================] - 82s 1ms/step - loss: 0.4430 - accuracy: 0.9165 - val_loss: 0.7546 - val_accuracy: 0.7648
Epoch 3/30
60792/60792 [==============================] - 83s 1ms/step - loss: 0.4348 - accuracy: 0.9208 - val_loss: 0.7417 - val_accuracy: 0.7693
Epoch 4/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4311 - accuracy: 0.9232 - val_loss: 0.7527 - val_accuracy: 0.7696
Epoch 5/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4277 - accuracy: 0.9252 - val_loss: 0.7729 - val_accuracy: 0.7677
Epoch 6/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4256 - accuracy: 0.9263 - val_loss: 0.7677 - val_accuracy: 0.7729
Epoch 7/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4253 - accuracy: 0.9269 - val_loss: 0.7529 - val_accuracy: 0.7698
Epoch 8/30
60792/60792 [==============================] - 82s 1ms/step - loss: 0.4248 - accuracy: 0.9273 - val_loss: 0.7521 - val_accuracy: 0.7702
Epoch 9/30
60792/60792 [==============================] - 80s 1ms/step - loss: 0.4226 - accuracy: 0.9282 - val_loss: 0.7706 - val_accuracy: 0.7676
Epoch 10/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4232 - accuracy: 0.9284 - val_loss: 0.7779 - val_accuracy: 0.7663
Epoch 11/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4227 - accuracy: 0.9282 - val_loss: 0.7851 - val_accuracy: 0.7649
Epoch 12/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4211 - accuracy: 0.9293 - val_loss: 0.7736 - val_accuracy: 0.7712
Epoch 13/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4209 - accuracy: 0.9292 - val_loss: 0.7696 - val_accuracy: 0.7717
Epoch 14/30
60792/60792 [==============================] - 83s 1ms/step - loss: 0.4211 - accuracy: 0.9295 - val_loss: 0.7815 - val_accuracy: 0.7727
Epoch 15/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4195 - accuracy: 0.9302 - val_loss: 0.7680 - val_accuracy: 0.7742
Epoch 16/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4193 - accuracy: 0.9302 - val_loss: 0.7637 - val_accuracy: 0.7711
Epoch 17/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4191 - accuracy: 0.9303 - val_loss: 0.7863 - val_accuracy: 0.7673
Epoch 18/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4187 - accuracy: 0.9305 - val_loss: 0.7871 - val_accuracy: 0.7686
Epoch 19/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4187 - accuracy: 0.9305 - val_loss: 0.7730 - val_accuracy: 0.7802
Epoch 20/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4191 - accuracy: 0.9304 - val_loss: 0.7953 - val_accuracy: 0.7735
Epoch 21/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4194 - accuracy: 0.9304 - val_loss: 0.7711 - val_accuracy: 0.7762
Epoch 22/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4185 - accuracy: 0.9307 - val_loss: 0.7800 - val_accuracy: 0.7698
Epoch 23/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4183 - accuracy: 0.9308 - val_loss: 0.7747 - val_accuracy: 0.7706
Epoch 24/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4180 - accuracy: 0.9310 - val_loss: 0.8009 - val_accuracy: 0.7711
Epoch 25/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4193 - accuracy: 0.9305 - val_loss: 0.7877 - val_accuracy: 0.7783
Epoch 26/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4181 - accuracy: 0.9310 - val_loss: 0.8029 - val_accuracy: 0.7687
Epoch 27/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4185 - accuracy: 0.9308 - val_loss: 0.7975 - val_accuracy: 0.7675
Epoch 28/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4175 - accuracy: 0.9313 - val_loss: 0.7793 - val_accuracy: 0.7735
Epoch 29/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4179 - accuracy: 0.9311 - val_loss: 0.7798 - val_accuracy: 0.7735
Epoch 30/30
60792/60792 [==============================] - 81s 1ms/step - loss: 0.4182 - accuracy: 0.9309 - val_loss: 0.7894 - val_accuracy: 0.7724
-------------------------------------------------------------------------------
----------------- THIS DATA WILL PROBABLY NOT BE USED -------------------------
Final accuracies for the different datasets using validation set number 3 after a total of 30 epoch(s).
* Accuracy on training set: 93.29%
* Accuracy on test set 1: 92.74%
* Accuracy on test set 2: 61.09%
* Accuracy on test set 3: 77.24%
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
</p>

<p>
    
    input_trainset:  (33149, 28, 28, 1)
target_trainset:  (33149,)
input_testset1:  (8288, 28, 28, 1)
target_testset1:  (8288,)
input_testset2:  (28563, 28, 28, 1)
target_testset2:  (28563,)
input_testset3:  (36851, 28, 28, 1)
target_testset3:  (36851,)
--------------------
training_pairs:  (60972, 2, 28, 28, 1)
training_target:  (60972,)
--------------------
test_pairs_set1:  (14760, 2, 28, 28, 1)
test_target_set1:  (14760,)
--------------------
test_pairs_set2:  (54592, 2, 28, 28, 1)
test_target_set2:  (54592,)
--------------------
test_pairs_set3:  (24600, 2, 28, 28, 1)
test_target_set3:  (24600,)
--------------------
Train on 60972 samples, validate on 14760 samples
Epoch 1/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.5735 - accuracy: 0.8392 - val_loss: 0.4458 - val_accuracy: 0.9167
Epoch 2/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4350 - accuracy: 0.9203 - val_loss: 0.4441 - val_accuracy: 0.9165
Epoch 3/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4270 - accuracy: 0.9243 - val_loss: 0.4397 - val_accuracy: 0.9203
Epoch 4/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4241 - accuracy: 0.9263 - val_loss: 0.4359 - val_accuracy: 0.9225
Epoch 5/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4224 - accuracy: 0.9270 - val_loss: 0.4324 - val_accuracy: 0.9249
Epoch 6/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4195 - accuracy: 0.9288 - val_loss: 0.4322 - val_accuracy: 0.9247
Epoch 7/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4190 - accuracy: 0.9290 - val_loss: 0.4294 - val_accuracy: 0.9262
Epoch 8/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4187 - accuracy: 0.9290 - val_loss: 0.4293 - val_accuracy: 0.9266
Epoch 9/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4173 - accuracy: 0.9300 - val_loss: 0.4318 - val_accuracy: 0.9252
Epoch 10/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4158 - accuracy: 0.9306 - val_loss: 0.4314 - val_accuracy: 0.9252
Epoch 11/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4158 - accuracy: 0.9306 - val_loss: 0.4312 - val_accuracy: 0.9253
Epoch 12/30
60972/60972 [==============================] - 79s 1ms/step - loss: 0.4163 - accuracy: 0.9306 - val_loss: 0.4340 - val_accuracy: 0.9245
Epoch 13/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4155 - accuracy: 0.9310 - val_loss: 0.4299 - val_accuracy: 0.9264
Epoch 14/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4162 - accuracy: 0.9307 - val_loss: 0.4275 - val_accuracy: 0.9278
Epoch 15/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4146 - accuracy: 0.9314 - val_loss: 0.4288 - val_accuracy: 0.9270
Epoch 16/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4151 - accuracy: 0.9315 - val_loss: 0.4284 - val_accuracy: 0.9278
Epoch 17/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4151 - accuracy: 0.9314 - val_loss: 0.4290 - val_accuracy: 0.9266
Epoch 18/30
60972/60972 [==============================] - 79s 1ms/step - loss: 0.4149 - accuracy: 0.9316 - val_loss: 0.4282 - val_accuracy: 0.9278
Epoch 19/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4143 - accuracy: 0.9319 - val_loss: 0.4312 - val_accuracy: 0.9257
Epoch 20/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4144 - accuracy: 0.9318 - val_loss: 0.4330 - val_accuracy: 0.9257
Epoch 21/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4138 - accuracy: 0.9318 - val_loss: 0.4299 - val_accuracy: 0.9261
Epoch 22/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4145 - accuracy: 0.9315 - val_loss: 0.4297 - val_accuracy: 0.9266
Epoch 23/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4141 - accuracy: 0.9319 - val_loss: 0.4310 - val_accuracy: 0.9264
Epoch 24/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4154 - accuracy: 0.9313 - val_loss: 0.4294 - val_accuracy: 0.9264
Epoch 25/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4147 - accuracy: 0.9315 - val_loss: 0.4306 - val_accuracy: 0.9260
Epoch 26/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4137 - accuracy: 0.9321 - val_loss: 0.4301 - val_accuracy: 0.9270
Epoch 27/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4138 - accuracy: 0.9320 - val_loss: 0.4292 - val_accuracy: 0.9269
Epoch 28/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4138 - accuracy: 0.9322 - val_loss: 0.4289 - val_accuracy: 0.9274
Epoch 29/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4139 - accuracy: 0.9321 - val_loss: 0.4285 - val_accuracy: 0.9275
Epoch 30/30
60972/60972 [==============================] - 78s 1ms/step - loss: 0.4137 - accuracy: 0.9321 - val_loss: 0.4302 - val_accuracy: 0.9268
-------------------------------------------------------------------------------
----------------- THIS DATA WILL PROBABLY NOT BE USED -------------------------
Final accuracies for the different datasets using validation set number 1 after a total of 30 epoch(s).
* Accuracy on training set: 93.38%
* Accuracy on test set 1: 92.68%
* Accuracy on test set 2: 68.25%
* Accuracy on test set 3: 80.96%
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
Train on 60972 samples, validate on 54592 samples
Epoch 1/30
60972/60972 [==============================] - 92s 2ms/step - loss: 0.5321 - accuracy: 0.8639 - val_loss: 0.9564 - val_accuracy: 0.6270
Epoch 2/30
60972/60972 [==============================] - 90s 1ms/step - loss: 0.4377 - accuracy: 0.9179 - val_loss: 0.8701 - val_accuracy: 0.6604
Epoch 3/30
60972/60972 [==============================] - 876s 14ms/step - loss: 0.4302 - accuracy: 0.9224 - val_loss: 0.9096 - val_accuracy: 0.6759
Epoch 4/30
60972/60972 [==============================] - 90s 1ms/step - loss: 0.4261 - accuracy: 0.9251 - val_loss: 1.1293 - val_accuracy: 0.5965
Epoch 5/30
60972/60972 [==============================] - 90s 1ms/step - loss: 0.4222 - accuracy: 0.9268 - val_loss: 0.9134 - val_accuracy: 0.6461
Epoch 6/30
60972/60972 [==============================] - 90s 1ms/step - loss: 0.4207 - accuracy: 0.9280 - val_loss: 1.1401 - val_accuracy: 0.5769
Epoch 7/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4203 - accuracy: 0.9283 - val_loss: 1.3293 - val_accuracy: 0.5547
Epoch 8/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4189 - accuracy: 0.9288 - val_loss: 1.0213 - val_accuracy: 0.6159
Epoch 9/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4188 - accuracy: 0.9291 - val_loss: 0.8909 - val_accuracy: 0.6620
Epoch 10/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4175 - accuracy: 0.9295 - val_loss: 0.9879 - val_accuracy: 0.6399
Epoch 11/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4164 - accuracy: 0.9305 - val_loss: 1.0851 - val_accuracy: 0.6033
Epoch 12/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4156 - accuracy: 0.9309 - val_loss: 1.1826 - val_accuracy: 0.5795
Epoch 13/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4161 - accuracy: 0.9306 - val_loss: 1.0039 - val_accuracy: 0.6273
Epoch 14/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4156 - accuracy: 0.9312 - val_loss: 1.1944 - val_accuracy: 0.5770
Epoch 15/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4154 - accuracy: 0.9311 - val_loss: 1.0369 - val_accuracy: 0.6144
Epoch 16/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4160 - accuracy: 0.9309 - val_loss: 1.0513 - val_accuracy: 0.6051
Epoch 17/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4151 - accuracy: 0.9312 - val_loss: 1.0179 - val_accuracy: 0.6236
Epoch 18/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4151 - accuracy: 0.9315 - val_loss: 1.0646 - val_accuracy: 0.6185
Epoch 19/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4156 - accuracy: 0.9312 - val_loss: 0.9600 - val_accuracy: 0.6466
Epoch 20/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4156 - accuracy: 0.9313 - val_loss: 1.1023 - val_accuracy: 0.6007
Epoch 21/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4149 - accuracy: 0.9314 - val_loss: 1.0586 - val_accuracy: 0.6204
Epoch 22/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4148 - accuracy: 0.9315 - val_loss: 1.0537 - val_accuracy: 0.6071
Epoch 23/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4147 - accuracy: 0.9317 - val_loss: 1.0639 - val_accuracy: 0.6199
Epoch 24/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4142 - accuracy: 0.9320 - val_loss: 1.0319 - val_accuracy: 0.6233
Epoch 25/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4150 - accuracy: 0.9317 - val_loss: 1.1105 - val_accuracy: 0.6135
Epoch 26/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4151 - accuracy: 0.9315 - val_loss: 1.0526 - val_accuracy: 0.6383
Epoch 27/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4147 - accuracy: 0.9319 - val_loss: 1.0592 - val_accuracy: 0.6443
Epoch 28/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4149 - accuracy: 0.9316 - val_loss: 1.1058 - val_accuracy: 0.6155
Epoch 29/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4138 - accuracy: 0.9322 - val_loss: 1.0367 - val_accuracy: 0.6385
Epoch 30/30
60972/60972 [==============================] - 91s 1ms/step - loss: 0.4146 - accuracy: 0.9319 - val_loss: 1.0173 - val_accuracy: 0.6420
-------------------------------------------------------------------------------
----------------- THIS DATA WILL PROBABLY NOT BE USED -------------------------
Final accuracies for the different datasets using validation set number 2 after a total of 30 epoch(s).
* Accuracy on training set: 93.38%
* Accuracy on test set 1: 92.28%
* Accuracy on test set 2: 64.20%
* Accuracy on test set 3: 78.60%
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
Train on 60972 samples, validate on 24600 samples
Epoch 1/30
60972/60972 [==============================] - 83s 1ms/step - loss: 0.5731 - accuracy: 0.8403 - val_loss: 0.7508 - val_accuracy: 0.7941
Epoch 2/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4439 - accuracy: 0.9162 - val_loss: 0.7016 - val_accuracy: 0.8099
Epoch 3/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4356 - accuracy: 0.9210 - val_loss: 0.7072 - val_accuracy: 0.8002
Epoch 4/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4327 - accuracy: 0.9230 - val_loss: 0.6900 - val_accuracy: 0.8073
Epoch 5/30
60972/60972 [==============================] - 81s 1ms/step - loss: 0.4284 - accuracy: 0.9251 - val_loss: 0.7109 - val_accuracy: 0.8051
Epoch 6/30
60972/60972 [==============================] - 81s 1ms/step - loss: 0.4275 - accuracy: 0.9259 - val_loss: 0.6926 - val_accuracy: 0.8176
Epoch 7/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4269 - accuracy: 0.9259 - val_loss: 0.7155 - val_accuracy: 0.7926
Epoch 8/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4256 - accuracy: 0.9272 - val_loss: 0.6978 - val_accuracy: 0.8079
Epoch 9/30
60972/60972 [==============================] - 81s 1ms/step - loss: 0.4241 - accuracy: 0.9281 - val_loss: 0.7108 - val_accuracy: 0.8103
Epoch 10/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4243 - accuracy: 0.9278 - val_loss: 0.7060 - val_accuracy: 0.8055
Epoch 11/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4232 - accuracy: 0.9283 - val_loss: 0.7111 - val_accuracy: 0.8120
Epoch 12/30
60972/60972 [==============================] - 83s 1ms/step - loss: 0.4239 - accuracy: 0.9279 - val_loss: 0.7129 - val_accuracy: 0.8148
Epoch 13/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4227 - accuracy: 0.9288 - val_loss: 0.7248 - val_accuracy: 0.8093
Epoch 14/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4215 - accuracy: 0.9292 - val_loss: 0.7171 - val_accuracy: 0.8095
Epoch 15/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4219 - accuracy: 0.9294 - val_loss: 0.7111 - val_accuracy: 0.8082
Epoch 16/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4212 - accuracy: 0.9295 - val_loss: 0.6965 - val_accuracy: 0.8166
Epoch 17/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4219 - accuracy: 0.9292 - val_loss: 0.7021 - val_accuracy: 0.8150
Epoch 18/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4208 - accuracy: 0.9299 - val_loss: 0.7081 - val_accuracy: 0.8112
Epoch 19/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4216 - accuracy: 0.9295 - val_loss: 0.7012 - val_accuracy: 0.8128
Epoch 20/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4215 - accuracy: 0.9293 - val_loss: 0.7086 - val_accuracy: 0.8166
Epoch 21/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4213 - accuracy: 0.9298 - val_loss: 0.6924 - val_accuracy: 0.8202
Epoch 22/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4207 - accuracy: 0.9301 - val_loss: 0.7456 - val_accuracy: 0.7945
Epoch 23/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4221 - accuracy: 0.9295 - val_loss: 0.7166 - val_accuracy: 0.8082
Epoch 24/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4207 - accuracy: 0.9300 - val_loss: 0.7069 - val_accuracy: 0.8173
Epoch 25/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4214 - accuracy: 0.9299 - val_loss: 0.7237 - val_accuracy: 0.8068
Epoch 26/30
60972/60972 [==============================] - 82s 1ms/step - loss: 0.4210 - accuracy: 0.9298 - val_loss: 0.7118 - val_accuracy: 0.8158
Epoch 27/30
60972/60972 [==============================] - 81s 1ms/step - loss: 0.4204 - accuracy: 0.9303 - val_loss: 0.7244 - val_accuracy: 0.8102
Epoch 28/30
60972/60972 [==============================] - 81s 1ms/step - loss: 0.4209 - accuracy: 0.9301 - val_loss: 0.7180 - val_accuracy: 0.8144
Epoch 29/30
60972/60972 [==============================] - 81s 1ms/step - loss: 0.4210 - accuracy: 0.9299 - val_loss: 0.7323 - val_accuracy: 0.8091
Epoch 30/30
60972/60972 [==============================] - 81s 1ms/step - loss: 0.4200 - accuracy: 0.9303 - val_loss: 0.7049 - val_accuracy: 0.8173
-------------------------------------------------------------------------------
----------------- THIS DATA WILL PROBABLY NOT BE USED -------------------------
Final accuracies for the different datasets using validation set number 3 after a total of 30 epoch(s).
* Accuracy on training set: 93.22%
* Accuracy on test set 1: 92.93%
* Accuracy on test set 2: 71.62%
* Accuracy on test set 3: 81.73%
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------

</p>


<p>
    
    input_trainset:  (33149, 28, 28, 1)
target_trainset:  (33149,)
input_testset1:  (8288, 28, 28, 1)
target_testset1:  (8288,)
input_testset2:  (28563, 28, 28, 1)
target_testset2:  (28563,)
input_testset3:  (36851, 28, 28, 1)
target_testset3:  (36851,)
--------------------
training_pairs:  (60912, 2, 28, 28, 1)
training_target:  (60912,)
--------------------
test_pairs_set1:  (14820, 2, 28, 28, 1)
test_target_set1:  (14820,)
--------------------
test_pairs_set2:  (54592, 2, 28, 28, 1)
test_target_set2:  (54592,)
--------------------
test_pairs_set3:  (24700, 2, 28, 28, 1)
test_target_set3:  (24700,)
--------------------
Train on 60912 samples, validate on 14820 samples
Epoch 1/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.5999 - accuracy: 0.8230 - val_loss: 0.4507 - val_accuracy: 0.9141
Epoch 2/30
60912/60912 [==============================] - 77s 1ms/step - loss: 0.4428 - accuracy: 0.9163 - val_loss: 0.4471 - val_accuracy: 0.9180
Epoch 3/30
60912/60912 [==============================] - 77s 1ms/step - loss: 0.4333 - accuracy: 0.9220 - val_loss: 0.4403 - val_accuracy: 0.9216
Epoch 4/30
60912/60912 [==============================] - 77s 1ms/step - loss: 0.4304 - accuracy: 0.9236 - val_loss: 0.4392 - val_accuracy: 0.9219
Epoch 5/30
60912/60912 [==============================] - 77s 1ms/step - loss: 0.4269 - accuracy: 0.9251 - val_loss: 0.4431 - val_accuracy: 0.9193
Epoch 6/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4254 - accuracy: 0.9260 - val_loss: 0.4376 - val_accuracy: 0.9240
Epoch 7/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4235 - accuracy: 0.9271 - val_loss: 0.4366 - val_accuracy: 0.9242
Epoch 8/30
60912/60912 [==============================] - 77s 1ms/step - loss: 0.4226 - accuracy: 0.9280 - val_loss: 0.4371 - val_accuracy: 0.9235
Epoch 9/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4225 - accuracy: 0.9280 - val_loss: 0.4360 - val_accuracy: 0.9237
Epoch 10/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4222 - accuracy: 0.9285 - val_loss: 0.4372 - val_accuracy: 0.9241
Epoch 11/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4210 - accuracy: 0.9288 - val_loss: 0.4373 - val_accuracy: 0.9240
Epoch 12/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4209 - accuracy: 0.9288 - val_loss: 0.4360 - val_accuracy: 0.9248
Epoch 13/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4194 - accuracy: 0.9297 - val_loss: 0.4346 - val_accuracy: 0.9252
Epoch 14/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4195 - accuracy: 0.9298 - val_loss: 0.4363 - val_accuracy: 0.9248
Epoch 15/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4192 - accuracy: 0.9300 - val_loss: 0.4380 - val_accuracy: 0.9238
Epoch 16/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4194 - accuracy: 0.9300 - val_loss: 0.4363 - val_accuracy: 0.9248
Epoch 17/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4194 - accuracy: 0.9299 - val_loss: 0.4369 - val_accuracy: 0.9243
Epoch 18/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4177 - accuracy: 0.9310 - val_loss: 0.4366 - val_accuracy: 0.9241
Epoch 19/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4183 - accuracy: 0.9304 - val_loss: 0.4365 - val_accuracy: 0.9237
Epoch 20/30
60912/60912 [==============================] - 98s 2ms/step - loss: 0.4187 - accuracy: 0.9303 - val_loss: 0.4369 - val_accuracy: 0.9247
Epoch 21/30
60912/60912 [==============================] - 79s 1ms/step - loss: 0.4181 - accuracy: 0.9306 - val_loss: 0.4379 - val_accuracy: 0.9242
Epoch 22/30
60912/60912 [==============================] - 79s 1ms/step - loss: 0.4182 - accuracy: 0.9307 - val_loss: 0.4367 - val_accuracy: 0.9248
Epoch 23/30
60912/60912 [==============================] - 79s 1ms/step - loss: 0.4179 - accuracy: 0.9309 - val_loss: 0.4387 - val_accuracy: 0.9238
Epoch 24/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4177 - accuracy: 0.9309 - val_loss: 0.4362 - val_accuracy: 0.9248
Epoch 25/30
60912/60912 [==============================] - 79s 1ms/step - loss: 0.4182 - accuracy: 0.9307 - val_loss: 0.4383 - val_accuracy: 0.9243
Epoch 26/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4187 - accuracy: 0.9303 - val_loss: 0.4414 - val_accuracy: 0.9221
Epoch 27/30
60912/60912 [==============================] - 80s 1ms/step - loss: 0.4185 - accuracy: 0.9306 - val_loss: 0.4362 - val_accuracy: 0.9251
Epoch 28/30
60912/60912 [==============================] - 79s 1ms/step - loss: 0.4175 - accuracy: 0.9310 - val_loss: 0.4359 - val_accuracy: 0.9251
Epoch 29/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4177 - accuracy: 0.9311 - val_loss: 0.4354 - val_accuracy: 0.9250
Epoch 30/30
60912/60912 [==============================] - 78s 1ms/step - loss: 0.4169 - accuracy: 0.9310 - val_loss: 0.4381 - val_accuracy: 0.9236
-------------------------------------------------------------------------------
----------------- THIS DATA WILL PROBABLY NOT BE USED -------------------------
Final accuracies for the different datasets using validation set number 1 after a total of 30 epoch(s).
* Accuracy on training set: 93.20%
* Accuracy on test set 1: 92.36%
* Accuracy on test set 2: 68.85%
* Accuracy on test set 3: 80.65%
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
Train on 60912 samples, validate on 54592 samples
Epoch 1/30
60912/60912 [==============================] - 93s 2ms/step - loss: 0.6417 - accuracy: 0.8326 - val_loss: 0.9226 - val_accuracy: 0.6836
Epoch 2/30
60912/60912 [==============================] - 91s 1ms/step - loss: 0.5541 - accuracy: 0.8853 - val_loss: 0.9343 - val_accuracy: 0.6880
Epoch 3/30
60912/60912 [==============================] - 91s 1ms/step - loss: 0.5465 - accuracy: 0.8905 - val_loss: 0.8690 - val_accuracy: 0.7174
Epoch 4/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5410 - accuracy: 0.8940 - val_loss: 0.9227 - val_accuracy: 0.7077
Epoch 5/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5395 - accuracy: 0.8946 - val_loss: 0.9339 - val_accuracy: 0.6966
Epoch 6/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5375 - accuracy: 0.8958 - val_loss: 0.8893 - val_accuracy: 0.7002
Epoch 7/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5359 - accuracy: 0.8968 - val_loss: 0.9009 - val_accuracy: 0.7119
Epoch 8/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5343 - accuracy: 0.8975 - val_loss: 0.8337 - val_accuracy: 0.7264
Epoch 9/30
60912/60912 [==============================] - 98s 2ms/step - loss: 0.5335 - accuracy: 0.8978 - val_loss: 0.9258 - val_accuracy: 0.7120
Epoch 10/30
60912/60912 [==============================] - 92s 2ms/step - loss: 0.5336 - accuracy: 0.8981 - val_loss: 0.8188 - val_accuracy: 0.7423
Epoch 11/30
60912/60912 [==============================] - 91s 1ms/step - loss: 0.5325 - accuracy: 0.8984 - val_loss: 0.8689 - val_accuracy: 0.7097
Epoch 12/30
60912/60912 [==============================] - 91s 1ms/step - loss: 0.5327 - accuracy: 0.8985 - val_loss: 0.9768 - val_accuracy: 0.7012
Epoch 13/30
60912/60912 [==============================] - 91s 1ms/step - loss: 0.5328 - accuracy: 0.8981 - val_loss: 0.8972 - val_accuracy: 0.7203
Epoch 14/30
60912/60912 [==============================] - 91s 1ms/step - loss: 0.5328 - accuracy: 0.8985 - val_loss: 0.8995 - val_accuracy: 0.7146
Epoch 15/30
60912/60912 [==============================] - 91s 1ms/step - loss: 0.5321 - accuracy: 0.8987 - val_loss: 0.8518 - val_accuracy: 0.7390
Epoch 16/30
60912/60912 [==============================] - 91s 1ms/step - loss: 0.5312 - accuracy: 0.8993 - val_loss: 0.8489 - val_accuracy: 0.7314
Epoch 17/30
60912/60912 [==============================] - 91s 1ms/step - loss: 0.5321 - accuracy: 0.8988 - val_loss: 0.8420 - val_accuracy: 0.7226
Epoch 18/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5317 - accuracy: 0.8992 - val_loss: 0.9145 - val_accuracy: 0.7124
Epoch 19/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5314 - accuracy: 0.8993 - val_loss: 0.9398 - val_accuracy: 0.7046
Epoch 20/30
60912/60912 [==============================] - 91s 1ms/step - loss: 0.5315 - accuracy: 0.8993 - val_loss: 0.8863 - val_accuracy: 0.7209
Epoch 21/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5306 - accuracy: 0.8995 - val_loss: 0.9026 - val_accuracy: 0.7237
Epoch 22/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5308 - accuracy: 0.8997 - val_loss: 0.9168 - val_accuracy: 0.7241
Epoch 23/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5303 - accuracy: 0.8999 - val_loss: 0.8913 - val_accuracy: 0.7275
Epoch 24/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5309 - accuracy: 0.8997 - val_loss: 0.9286 - val_accuracy: 0.7226
Epoch 25/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5311 - accuracy: 0.8995 - val_loss: 0.8923 - val_accuracy: 0.7277
Epoch 26/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5313 - accuracy: 0.8997 - val_loss: 0.8809 - val_accuracy: 0.7284
Epoch 27/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5310 - accuracy: 0.8995 - val_loss: 0.8631 - val_accuracy: 0.7332
Epoch 28/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5301 - accuracy: 0.9002 - val_loss: 0.9285 - val_accuracy: 0.6992
Epoch 29/30
60912/60912 [==============================] - 91s 1ms/step - loss: 0.5299 - accuracy: 0.9001 - val_loss: 0.8993 - val_accuracy: 0.7256
Epoch 30/30
60912/60912 [==============================] - 90s 1ms/step - loss: 0.5306 - accuracy: 0.9001 - val_loss: 0.8110 - val_accuracy: 0.7506
-------------------------------------------------------------------------------
----------------- THIS DATA WILL PROBABLY NOT BE USED -------------------------
Final accuracies for the different datasets using validation set number 2 after a total of 30 epoch(s).
* Accuracy on training set: 90.19%
* Accuracy on test set 1: 89.24%
* Accuracy on test set 2: 75.06%
* Accuracy on test set 3: 80.38%
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
Train on 60912 samples, validate on 24700 samples
Epoch 1/30
60912/60912 [==============================] - 83s 1ms/step - loss: 0.5745 - accuracy: 0.8379 - val_loss: 0.7148 - val_accuracy: 0.7928
Epoch 2/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4420 - accuracy: 0.9167 - val_loss: 0.7164 - val_accuracy: 0.8011
Epoch 3/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4336 - accuracy: 0.9216 - val_loss: 0.7027 - val_accuracy: 0.8086
Epoch 4/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4291 - accuracy: 0.9244 - val_loss: 0.7018 - val_accuracy: 0.8101
Epoch 5/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4264 - accuracy: 0.9256 - val_loss: 0.7002 - val_accuracy: 0.8165
Epoch 6/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4250 - accuracy: 0.9262 - val_loss: 0.7002 - val_accuracy: 0.8185
Epoch 7/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4234 - accuracy: 0.9275 - val_loss: 0.7114 - val_accuracy: 0.8167
Epoch 8/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4214 - accuracy: 0.9286 - val_loss: 0.7117 - val_accuracy: 0.8115
Epoch 9/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4209 - accuracy: 0.9290 - val_loss: 0.7343 - val_accuracy: 0.7948
Epoch 10/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4208 - accuracy: 0.9286 - val_loss: 0.7119 - val_accuracy: 0.8020
Epoch 11/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4197 - accuracy: 0.9298 - val_loss: 0.7249 - val_accuracy: 0.8081
Epoch 12/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4196 - accuracy: 0.9296 - val_loss: 0.7445 - val_accuracy: 0.8017
Epoch 13/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4196 - accuracy: 0.9301 - val_loss: 0.7324 - val_accuracy: 0.8041
Epoch 14/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4185 - accuracy: 0.9302 - val_loss: 0.7418 - val_accuracy: 0.8044
Epoch 15/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4186 - accuracy: 0.9305 - val_loss: 0.7615 - val_accuracy: 0.7970
Epoch 16/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4182 - accuracy: 0.9301 - val_loss: 0.7401 - val_accuracy: 0.8047
Epoch 17/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4191 - accuracy: 0.9300 - val_loss: 0.7625 - val_accuracy: 0.7960
Epoch 18/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4181 - accuracy: 0.9307 - val_loss: 0.7369 - val_accuracy: 0.8051
Epoch 19/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4191 - accuracy: 0.9301 - val_loss: 0.7235 - val_accuracy: 0.8099
Epoch 20/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4181 - accuracy: 0.9306 - val_loss: 0.7423 - val_accuracy: 0.7995
Epoch 21/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4189 - accuracy: 0.9304 - val_loss: 0.7417 - val_accuracy: 0.8001
Epoch 22/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4182 - accuracy: 0.9304 - val_loss: 0.7360 - val_accuracy: 0.8071
Epoch 23/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4178 - accuracy: 0.9308 - val_loss: 0.7455 - val_accuracy: 0.8036
Epoch 24/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4178 - accuracy: 0.9307 - val_loss: 0.7360 - val_accuracy: 0.8048
Epoch 25/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4176 - accuracy: 0.9309 - val_loss: 0.7292 - val_accuracy: 0.8086
Epoch 26/30
60912/60912 [==============================] - 80s 1ms/step - loss: 0.4171 - accuracy: 0.9309 - val_loss: 0.7275 - val_accuracy: 0.8140
Epoch 27/30
60912/60912 [==============================] - 81s 1ms/step - loss: 0.4175 - accuracy: 0.9312 - val_loss: 0.7452 - val_accuracy: 0.8023
Epoch 28/30
60912/60912 [==============================] - 80s 1ms/step - loss: 0.4189 - accuracy: 0.9304 - val_loss: 0.7375 - val_accuracy: 0.8045
Epoch 29/30
60912/60912 [==============================] - 80s 1ms/step - loss: 0.4168 - accuracy: 0.9312 - val_loss: 0.7464 - val_accuracy: 0.8028
Epoch 30/30
60912/60912 [==============================] - 80s 1ms/step - loss: 0.4171 - accuracy: 0.9313 - val_loss: 0.7250 - val_accuracy: 0.8123
-------------------------------------------------------------------------------
----------------- THIS DATA WILL PROBABLY NOT BE USED -------------------------
Final accuracies for the different datasets using validation set number 3 after a total of 30 epoch(s).
* Accuracy on training set: 93.32%
* Accuracy on test set 1: 92.59%
* Accuracy on test set 2: 70.26%
* Accuracy on test set 3: 81.23%
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------

</p>